lib/dao.js

```diff
@@ -1597,6 +1598,22 @@ DataAccessObject._coerce = function(where) {
 
       continue;
     }
+
+    if (p.match(/\./)) {
+      var model = p.split('.')[0];
+      var prop = p.split('.').slice(1);
+
+      if (props[model]) {
+        var clause = {};
+        clause[prop] = where[p];
+        where[p] = Array.isArray(props[model].type) ?
+          self._coerce(clause, props[model].type[0].definition.properties)[prop] :
+          self._coerce(clause, props[model].type.definition.properties)[prop];
```

Comment:
@yshing hello, thank you for reporting the problem. How can we reproduce it ourselves, so that we can add a unit-test to prevent a regression like this re-appearing in the future?

I mean how does it happened that `...type.definition` does not exist?

BTW considering that it's clear where and how to fix the problem, would you like to contribute this fix yourself? See http://loopback.io/doc/en/contrib/index.html and http://www.nearform.com/nodecrunch/first-time-with-open-source/ to get you started. Just make sure to include a unit-test that's failing with the current version and starts passing with your fix in place.